src/runtime/Temporal.h

```diff
@@ -37,9 +39,49 @@ class Temporal : public DerivedObject {
         return true;
     }
 
-    bool hasInitializedTemporalDate() const
+    static Value toTemporalOverflow(ExecutionState& state, const Value& normalizedOptions)
     {
-        return false;
+        if (normalizedOptions.isUndefined()) {
+            return state.context()->staticStrings().lazyConstrain().string();
+        }
+        Value matcherValues[2] = { state.context()->staticStrings().lazyConstrain().string(), state.context()->staticStrings().reject.string() };
+        return Intl::getOption(state, normalizedOptions.asObject(), state.context()->staticStrings().lazyoverflow().string(), Intl::StringValue, matcherValues, 2, matcherValues[0]);
+    }
+
+    static Value prepareTemporalFields(ExecutionState& state, const Value& fields, const ValueVector& fieldNames, const ValueVector& requiredFields)
+    {
+        ASSERT(fields.isObject());
+        auto* result = new Object(state);
+        std::string temporalFields[14] = { "year", "month", "monthCode", "day", "hour", "minute", "second", "millisecond",
+                                           "microsecond", "nanosecond", "offset", "era", "eraYear", "timeZone" };
+        for (auto property : fieldNames) {
+            Value value;
+            value = fields.asObject()->get(state, ObjectPropertyName(state, fields), property).value(state, Value());
+            if (value.isUndefined()) {
+                if (std::find(requiredFields.begin(), requiredFields.end(), property) != requiredFields.end()) {
+                    ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, new ASCIIString("requiredFields contains property"));
+                }
+                std::string prop = property.asString()->toNonGCUTF8StringData();
+                if (prop == temporalFields[0] || prop == temporalFields[1] || prop == temporalFields[2] || prop == temporalFields[3] || prop == temporalFields[10] || prop == temporalFields[11] || prop == temporalFields[12] || prop == temporalFields[13]) {
+                    value = Value();
+                } else {
+                    value = Value(0);
+                }
+            } else {
+                std::string prop = property.asString()->toNonGCUTF8StringData();
+                if (prop != temporalFields[13]) {
+                    if (prop == temporalFields[2] || prop == temporalFields[10] || prop == temporalFields[11]) {
+                        value = Value(value.toString(state));
+                    } else if (prop == temporalFields[1] || prop == temporalFields[3]) {
+                        value = Value(value.toUint32(state));
+                    } else {
+                        value = Value(value.toInteger(state));
```

Comment:
I cannot find any usage case of this `value` variable after here.
Is this `value` calculation really necessary?

Reply:
In line `81` I called the defineOwnProperty method wrong and the this value should be a parameter of this method.